ludwig/data/dataframe/dask.py

```diff
@@ -43,15 +44,15 @@ def __init__(self, parallelism=None, persist=True, **kwargs):
     def set_parallelism(self, parallelism):
         self._parallelism = parallelism
 
-    def df_like(self, df, proc_cols):
+    def df_like(self, df: dd.DataFrame, proc_cols: Dict[str, dd.Series]):
         # Our goal is to preserve the index of the input dataframe but to drop
         # all its columns. Because to_frame() creates a column from the index,
         # we need to drop it immediately following creation.
         dataset = df.index.to_frame(name=TMP_COLUMN).drop(columns=[TMP_COLUMN])
         # TODO: address if following results in fragmented DataFrame
-        for k, v in proc_cols.items():
-            dataset[k] = v
-            dataset[k] = dataset[k].astype(v.dtype)
+        col_names, cols = zip(*proc_cols.items())
+        dataset = dd.concat([dataset] + list(cols), axis=1)
```

Comment:
If this preserves the cols datatype as i imagine it looks like an interesting workaround for preserving the index and not having to do the assignment and casting explicitly :)

Reply:
+1

Reply:
This may also help with the large-number-of-columns issue we see using the other approach.